docs/webiny-apps/page-builder/development/plugins-reference/app.md

```diff
@@ -0,0 +1,185 @@
+    create: (options: { [key: string]: any }, parent?: PbElement) => Omit<PbElement, "id" | "path">;
+    // A function to render an element in the editor.
+    render: (params: { theme?: PbTheme; element: PbElement }) => ReactNode;
+    // A function to check if an element can be deleted.
+    canDelete?: (params: { element: PbElement }) => boolean;
+    // Executed when another element is dropped on the drop zones of current element.
+    onReceived?: (params: {
+        store?: Store;
+        source: PbElement | { type: string; path?: string };
+        target: PbElement;
+        position: number | null;
+    }) => void;
+    // Executed when an immediate child element is deleted
+    onChildDeleted?: (params: { element: PbElement; child: PbElement }) => void;
+    // Executed after element was created
+    onCreate?: string;
+    // Render element preview (used when creating element screenshots; not all elements have a simple DOM representation
+    // so this callback is used to customize the look of the element in a PNG image)
+    renderElementPreview?: (params: {
+        element: PbElement;
+        width: number;
+        height: number;
+    }) => ReactElement;
+};
+```
+
+#### Example
+
+```ts
+{
+    name: "pb-editor-page-element-iframe",
+    type: "pb-editor-page-element",
+    elementType: "iframe",
+    toolbar: {
+        title: "iFrame",
+        group: "pb-editor-element-group-media",
+        preview() {
+            return (
+                <PreviewBox>
+                    <IFrameIcon />
+                </PreviewBox>
+            );
+        }
+    },
+    settings: ["pb-editor-page-element-settings-delete"],
+    onCreate: "open-settings",
+    create(options) {
+        return {
+            type: "iframe",
+            elements: [],
+            data: {
+                iframe: {
+                    url: "",
+                    height: 370
+                },
+                settings: {
+                    horizontalAlign: "center",
+                    margin: {
+                        desktop: { all: 0 },
+                        mobile: { all: 0 }
+                    },
+                    padding: {
+                        desktop: { all: 0 },
+                        mobile: { all: 0 }
+                    }
+                }
+            },
+            ...options
+        };
+    },
+    render(props) {
+        return <IFrameEmbed {...props} />;
+    },
+    renderElementPreview({ width, height }) {
+        return <img style={{ width, height }} alt={"iFrame"} />;
+    }
+}
+```
+
+![Editor Iframe Plugin](/img/webiny-apps/page-builder/development/development/plugin-reference/editor/iframe/editor-iframe-plugin.png)
+
+
+### [`pb-editor-page-element-advanced-settings`](/docs/webiny-apps/page-builder/development/plugins-reference/app#pb-editor-page-element-advanced-settings)
+
+#### Summary
+
+Enables adding custom settings dialog fields into the page builder.
```

Comment:
`Enables adding custom settings dialog fields into the page builder` => `Enables defining custom settings dialogs for page elements.`

Also, add a tip a short sentence which mentions how these are consumed by the `PbEditorPageElementPlugin` plugin.

Reply:
I added this sentence: ```The `PbEditorPageElementPlugin` will consume the data that is bound up using the Bind component below.```